### ISSUE TITLE
Wrapped Scripting.Dictionary: Error when accessing an Item of the arrays from the Keys or Items Methods.

### ISSUE THREAD

**Issue by FullValueRider** [WaynePhillipsEA/twinbasic#720]
**Describe the bug**
Thanks to code provided by @Kr00l I now have a class (Enumerable) which can enumerate any enumerable object or string forwards, backwards, or  partially, and for dictionaries  by key, item or key/item pairs.  However I have encountered a problem that when I supply a scripting.dictionary the code crashes as indicated below.  If I extract the Items or Keys of the scripting dictionary into their own variant I can access items fine (as I have done for debugging purposes) , as demonstrated by the code below.

P.input is the object supplied to the class
P.Items holds a reference to p.Input and allows for the case where p.Input has to be transformed into an array for access by index.
```
 Case IsDictionary
        
            Dim myKeys As Variant
            myKeys = p.Items.keys
            
            Dim myitems As Variant
            myitems = p.Items.items
            
            Select Case p.DictionaryOutput
                
                Debug.Print Index + p.IndexOffset  ' should this line get executed.
                
                Case enumDIctionaryOutput.Value(asKeys)
                    Debug.Print myKeys(Index + p.IndexOffset)
                     'myresult = Array(p.Items.keys(Index + p.IndexOffset))     'This line fails and causes a jump to IUnknown_Release in ModEnumVariant
                                                                                ' using just an SD (as in the TestSD method in module Pootle module works OK 
                    myresult = Array(myKeys(Index + p.IndexOffset))              'This line works fine
                   'myresult = Array(p.Input.keys(Index + p.IndexOffset))       ' This line also fails in the same way as p.Input
                
                Case enumDIctionaryOutput.Value(asItems)
                
                    myresult = Array(myitems(myKeys(Index + p.IndexOffset)))
                    
                
                Case enumDIctionaryOutput.Value(asPairs)
                
                    myresult = Array(Array(myKeys(Index + p.IndexOffset), myitems(myKeys(Index + p.IndexOffset))))
                
                
            End Select

```
The project is attached below and contains Tests to show things work fine for non enumerable items, strings, 1D array (List arrays), collection, arraylist,stack and queue.  The fact that the Keys and Items methods work shows that the Dictionary exists, its just that Keys(Index) and Items(index) don't work.

Please run the tests using the Test.Tester Method and not unit testing pane.

[Enumerable.zip](https://github.com/WaynePhillipsEA/twinbasic/files/8027542/Enumerable.zip)

**Expected behavior**
p.Items.Keys(Index+P.IndexOffset) should not cause an error when Index+P.IndexOffset evaluate to a legal index.

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 11]
 - twinBASIC compiler version [e.g. 0.14.9]

**Comment by WaynePhillipsEA** [comment]
Running `Test.Tester` in v0.14.10 results in no error/crash.  Please confirm if the issue is fixed? (probably due to #190 being fixed)

**Comment by FullValueRider** [comment]
Apologies for the lack of clarity. 

In Enumerable.Item (Property Get)  Case IsDictionary  the code as supplied is
```
  'myresult = Array(p.Items.keys(Index + p.IndexOffset))     'This line fails and causes a jump to IUnknown_Release in ModEnumVariant
                                                                                ' using just an SD (as in the TestSD method in module Pootle module works OK 
   myresult = Array(myKeys(Index + p.IndexOffset))              'This line works fine
```
If I change it to
```
     myresult = Array(p.Items.keys(Index + p.IndexOffset))     'This line fails and causes a jump to IUnknown_Release in ModEnumVariant
                                                                                ' using just an SD (as in the TestSD method in module Pootle module works OK 
    'myresult = Array(myKeys(Index + p.IndexOffset))              'This line works fine
```
Then the error still occurs in 14.10
